yfinance/base.py

```diff
@@ -512,11 +512,16 @@ def get_news(self, proxy=None) -> list:
         # Getting data from json
         url = f"{_BASE_URL_}/v1/finance/search?q={self.ticker}"
         data = self._data.cache_get(url=url, proxy=proxy)
-        if "Will be right back" in data.text:
+        if data is None or "Will be right back" in data.text:
             raise RuntimeError("*** YAHOO! FINANCE IS CURRENTLY DOWN! ***\n"
                                "Our engineers are working quickly to resolve "
                                "the issue. Thank you for your patience.")
-        data = data.json()
+        try:
+            data = data.json()
+        except (_json.JSONDecodeError): 
+            logger = utils.get_yf_logger()
+            logger.error(f"{self.ticker}: Failed to retireve the news and received faulty response instead.")
```

Comment:
`retireve` typo

Reply:
Thanks. Addressed in: https://github.com/ranaroussi/yfinance/pull/2021/commits/3574cf4bdc7706ce0dcf87cd4f67b16e781c2217